Objects/StrategyCardAutomator.ttslua

```diff
@@ -0,0 +1,1648 @@
+		broadcastToAll('Assign speaker token: no faction selected for ' .. color, 'Red')
+		return
+	end
+
+	local speakerToken = nil
+	for _, object in ipairs(getAllObjects()) do
+		local name = object.getName()
+		local guid = object.getGUID()
+		if name == "Speaker Token" then
+			speakerToken = object
+			break
+		end
+	end
+
+
+	local commandSheet = getObjectFromGUID(faction.commandSheetGuid)
+	if not commandSheet then
+		broadcastToAll('Assign speaker token: missing ' .. color .. ' Command Sheet', 'Red')
+		return
+	end
+	local rot = commandSheet.getRotation()
+	zone = _zoneHelper.zoneAttributes(color)
+
+	local loc = _getSpeakerLoc(zone)
+
+	speakerToken.setPositionSmooth( loc, false, true )
+	speakerToken.setRotation(rot)
+
+	broadcastToAll(color .. " has been assigned the new speaker.", color)
+	return true
+end
+
+-- Get corner of zone closest to the system.
+-- also tested for 7 and 8 players
+function _getSpeakerLoc(zone)
+	local polygons = zone.polygon
+
+	-- get any z value closest to the system map.
+	local minz = math.abs(polygons[1][2])
+	local i_minz = 1
+	for k,poly in ipairs(polygons) do
+		if math.abs(poly[2]) < minz then
+			minz = math.abs(poly[2])
+			i_minz = k
+		end
+	end
+	local z = polygons[i_minz][2]
+
+	-- get the x value closest to the system map, with the matching z value.
+	local minx = polygons[i_minz][1]
+	local i_minx = 1
+	for k,poly in ipairs(polygons) do
+		if (z == poly[2]) and (math.abs(poly[1]) <= minx) then
+			minx = math.abs(poly[1])
+			i_minx = k
+		end
+	end
+	local x = polygons[i_minx][1]
+
+	-- shift the token slightly into the zone.
+	if z > 0 then z = z + 3 else z = z - 3 end
+	if x > 0 then x = x + 4 else x = x - 4 end
+
+	return {x=x ,y=2, z=z}
+end
+
+-- option can be "self" or target "COLOR".
+function _assignSpeaker(color, option)
+	local res
+	if option=="self" then
+		res = _moveSpeakerToken(color)
+	else
+		res = _moveSpeakerToken(option)
+	end
+	return res
+end
+
+function _dealPoliticsActionCards(color)
+	-- log("dealing politics to " .. color)
+	local counter = 2
+
+	local faction = _factionHelper.fromColor(color)
+	if not faction then
+		broadcastToAll('Allocate command tokens: no faction for ' .. color, 'Red')
+		return
+	end
+
+	local abilitiesSet = {}
+	for _, ability in ipairs(faction.abilities) do
+
+		abilitiesSet[ability] = true
+	end
+	if abilitiesSet['Scheming'] then
+		broadcastToAll("Scheming.", color)
+		counter = counter + 1
+	end
+
+	_deckHelper.deal({
+		deck = 'Actions',
+		count = counter,
+		color = color,
+		quiet = true
+	})
+end
+
+
+function _dealPoliticsAgendas(color)
+	_deckHelper.deal({
+		deck = 'Agenda',
+		count = 2,
+		color = color,
+		quiet = true
+	})
+
+	_safeBroadcastToColor("You have drawn 2 agendas.", color, color)
+end
+
+------------------------------------ Trade -------------------------------------
+
+function _gain3Tradegoods(color)
+-- deals one 3xTradegoods token to color. Position on commandsheet is hardcoded.
+	local commoditiesBag = false
```

Comment:
I like to cache guids so future lookups don't scan the table.  since there are probably a few of these can make that a shared function: `
function _getByName(name)
    local guid = _getByNameCache[name]
    local object = guid and getObjectFromGUID(guid)
    if object then
        return object
    end
    for _, object in ipairs(getAllObjects()) do
        if object.getName() == name then
            _getByNameCache[name] = object.getGUID()
            return object
        end
    end
end
`
(could have a tag check too)